File: CriminalIntent/app/src/main/java/kg/dostek/criminalintent/CrimeActivity.java
package kg.dostek.criminalintent;

import android.content.Context;
import android.content.Intent;
import android.os.Bundle;
import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentActivity;
import android.support.v4.app.FragmentManager;

import java.util.UUID;


public class CrimeActivity extends SingleFragmentActivity {
    private static final String EXTRA_CRIME_ID = "kg.dostek.criminalintent.crime_id";
    public static final String EXTRA_VIEW_POSITION = "kg.dostek.criminalintent.view_position";

    public static Intent newIntent(Context packageContext, UUID crimeId, int position) {
        Intent intent = new Intent(packageContext, CrimeActivity.class);
        intent.putExtra(EXTRA_VIEW_POSITION, position);
        intent.putExtra(EXTRA_CRIME_ID, crimeId);
        return intent;
    }

    @Override
    protected Fragment createFragemnt() {
        UUID crimeId = (UUID) getIntent().getSerializableExtra(CrimeActivity.EXTRA_CRIME_ID);
        int position = getIntent().getIntExtra(CrimeActivity.EXTRA_VIEW_POSITION, -1);

        Intent intent = new Intent();
        intent.putExtra(EXTRA_VIEW_POSITION, position);
        setResult(RESULT_OK, intent);

        return CrimeFragment.newInstance(crimeId);
    }

    public static int getPosition(Intent data) {
        return data.getIntExtra(EXTRA_VIEW_POSITION, -1);
    }
}

File: CriminalIntent/app/src/main/kotlin/kt/akimov/criminalintent/presentation/crimeitem/CrimeActivity.kt
package kt.akimov.criminalintent.presentation.crimeitem

import android.content.Context
import android.content.Intent
import android.os.Bundle
import android.util.Log
import kt.akimov.criminalintent.presentation.SingleFragmentActivity
import java.util.*


class CrimeActivity : SingleFragmentActivity() {

    override fun createFragment() : CrimeFragment {
        val crimeId = intent?.getSerializableExtra(EXTRA_CRIME_ID) as UUID
        return CrimeFragment.newInstance(crimeId)
    }

    companion object {
        private val EXTRA_CRIME_ID = "kt.akimov.criminalintent.crime_id"
        fun newIntent(packageContext: Context?, crimeId: UUID): Intent {
            val intent = Intent(packageContext, CrimeActivity::class.java)
            intent.putExtra(EXTRA_CRIME_ID, crimeId)
            return intent
        }
    }

    /*override fun onCreate(savedInstanceState: Bundle?) {
		super.onCreate(savedInstanceState)
		Log.d(TAG, "onCreate")
	}

	override fun onStart() {
		super.onStart()
		Log.d(TAG, "onStart")
	}

	override fun onResume() {
		super.onResume()
		Log.d(TAG, "onResume")
	}

	override fun onPause() {
		super.onPause()
		Log.d(TAG, "onPause")
	}

	override fun onStop() {
		super.onStop()
		Log.d(TAG, "onStop")
	}

	override fun onDestroy() {
		super.onDestroy()
		Log.d(TAG, "onDestroy")
	}

	override fun onRestart() {
		super.onRestart()
		Log.d(TAG, "onRestart")
	}

	companion object {
		private var TAG: String = CrimeActivity::class.java.simpleName
	}*/
}